core/system/src/mpu/vmpu.c

```diff
@@ -317,6 +326,9 @@ int vmpu_fault_recovery_bus(uint32_t pc, uint32_t sp)
     /* note: we assume the instruction is 16 bits wide and skip possible NOPs */
     vmpu_unpriv_uint32_write(sp + (6 << 2), pc + ((UVISOR_NOP_CNT + 2 - cnt) << 1));
 
+    /* clear BFSR only - reset BFARVALID bit to acknowledge fault */
+    SCB_BFSR = 0x80;
```

Comment:
SCB_BFSR needs to be cleared even in case the function returns - we might want to consider in this function to clear it right after checking it at the top.